node/src/components/consensus/highway_core/highway.rs

```diff
@@ -304,6 +324,7 @@ impl<C: Context> Highway<C> {
             Dependency::Unit(hash) => self.state.has_unit(hash),
             Dependency::Evidence(idx) => self.state.is_faulty(*idx),
             Dependency::Endorsement(hash) => self.state.is_endorsed(hash),
+            Dependency::Ping(_, _) => false, // We don't store signatures; nothing depends on pings.
```

Comment:
I think the comment should say that `Ping` does not depend on anything.

Reply:
No, `has_dependency` is about asking whether we have a unit with a particular hash, etc.
If it asks us whether we have a particular ping, it just always returns false, because we don't even store the signatures and pings are never requested as dependencies.

All of this slightly abuses the vertices-and-dependencies system, but it avoids a lot of `Option`s or a whole new message type.